_posts/scyrill/11-23-2021-scyrill-project-update.md

```diff
@@ -0,0 +1,158 @@
+```python 
+#open and read the file 
+notes = open("MIMIC-IIIData.txt", "r")
+pt_notes = notes.readlines()
+pt_notes.close()
+print(pt_notes)
+```
+This didn’t print anything, to my frustration, until I figured out that the reason it wasn’t printing anything was because I closed the file, so I left it open and changed ```readlines()``` to ```read()```. Once I did this, the file printed everything again. From here, I moved on to putting everything into a list: 
+```python 
+#open and read the file 
+notes = open("MIMIC-IIIData.txt", "r")
+pt_notes = notes.read()
+diagnosis = pt_notes.split(“Discharge Diagnosis:”)
+print(diagnosis)
+```
+However, this is not what I wanted to do, so I asked my friend, and she told me that, as is, I am telling Python to print the entire file in a list and if I wanted to get python to grab everything after the first instance of ‘Discharge Diagnosis’, I would have to print that position, which resulted in me putting ```print(diagnosis[1])```, which wound return the first line after ‘Discharge Diagnosis’ which is the list of diagnoses that I wanted. The next step was to get Python to stop printing right before the phrase ‘Diagnosis Condition’. To do this, I added the following lines: 
+```python 
+dc = pt_notes.split(“Discharge Condition:”)
+print(dc[0])
+```
+This ended up returning all the lines before ‘Discharge Condition:’, which was partially what I wanted since it didn’t include the phrase ‘Discharge Condition:’, but it was not printing just the diagnoses that I wanted. Some learnings that I gained from this were that with ```dc[0]```, it completely cuts ‘Discharge Condition:’ off from the file, whereas the ```diagnosis[1]``` included what I wanted, so I had to find a way to put them together. I ended up making ```diagnosis[1]``` a variable and then splitting it in a new variable called ```dc```. To have a place to store all this information, I introduced an empty list called ```bank``` to put the diagnoses. From here, I added the diagnoses to ```bank``` by using the ```append()``` function, thus resulting in the following code: 
+```python 
+#open and read the file 
+notes = open("MIMIC-IIIData.txt", "r")
+pt_notes = notes.read()
+bank = []
+diagnosis = pt_notes.split(“Discharge Diagnosis:”)
+dd = diagnosis[1]
+dc = dd.split(“Discharge Condition:”)
+bank.append(dc[0])
+print(bank)
+```
+This finally worked! It returned all the diagnoses, but it included the whitespace character ```\n```, which was something I wanted to get rid of. I knew that I could use ```strip``` to take out the whitespace, so I tried using ```rstrip()``` to do so. I chose to do this by introducing another variable called ```entries```, attaching ```rstrip()```, and putting into what I wanted it to strip out of the list. 
```

Comment:
Again, the wine red highlighting makes me think something is getting mixed up here. Try just one backtick before and after the `\n` (and the other in-line code highlighted) and see if that clears things up.

In markdown you can use one backtick for in-line code and three for multiline code blocks.

Reply:
Got it, but this makes me wonder why this happened here and not other points in the code, like when I referred to regex or other areas?

Reply:
I added new commit addressing this issue, but things are still showing up in this wine red highlight, not sure why though

Reply:
I think you're right, that this is actually being caused by things happening elsewhere in this file. What happens if you replace the angle single `‘’` and double quotes `“”` with straight ones? As you notice, when you're doing things like writing regex expressions within code blocks they may be causing problems.

Reply:
I straightened them out but the same issue is happening in the same place

Reply:
Looks to me like your most recent changes fix things. Now when I look at the [files changed](https://github.com/inf380p/inf380p.github.io/pull/563/files) on this pull request the wine red syntax highlighting is gone. Does that look right to you?

Thanks for working through this with me!